### PR TITLE
Update head rev to die on mind-shield

### PR DESCRIPTION
**Pull request by Titian3**
## About the PR
Head revs die on mind-shielding, they bleed out and suffer unrevivable damage.

## Why / Balance
Admins are discussing to add a new rule to prevent head revs being held on to by other crew that prolong rounds and stall them out by the head revs not being executed. 
This would mechanically prevent those issues and prevent another rule from being added.

## Technical details
On mind shielding for a head rev, the entity takes 300 genetic damage and looses all their blood out to the floor in a simulated hemorrhage.

## Media

https://github.com/space-wizards/space-station-14/assets/47093363/3c64e8a2-8410-43f8-afdf-ccd28e2b1ba7

image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
None

**Changelog**
:cl: Repo
- tweak: Head revs now die on mind shielding.

### CODE REVIEW

**Comment by EmoGarbage404** [comment]
my main concern is that this is an incredibly quick way to kill someone that is not communicated very well to the player. Instantly killing the player after a 3(?) second doafter is really extreme, especially for an antag. I could very much see these being used as a way to try and quick-kill revs. Stunning someone with a baton and quickly injecting while they're down could be seriously overpowered if it just straight up kills headrevs.

**Comment by Hmeister-fake** [comment]
> my main concern is that this is an incredibly quick way to kill someone that is not communicated very well to the player. Instantly killing the player after a 3(?) second doafter is really extreme, especially for an antag. I could very much see these being used as a way to try and quick-kill revs. Stunning someone with a baton and quickly injecting while they're down could be seriously overpowered if it just straight up kills headrevs.

Agreed, straight up round ending headrevs after a 5  second do-after is absurd. The only way you could realistically round end someone that fast normally is with a christmas gib stick and eating the brain right after :skull:

File: Content.Server/Mindshield/MindShieldSystem.cs [hunk 1]
```diff
@@ -52,6 +61,17 @@ public void MindShieldRemovalCheck(EntityUid implanted, EntityUid implant)
         {
             _popupSystem.PopupEntity(Loc.GetString("head-rev-break-mindshield"), implanted);
             QueueDel(implant);
+            if (TryComp<BloodstreamComponent>(implanted, out var bloodstream))
+            {
+                _bloodstreamSystem.SpillAllSolutions(implanted, bloodstream);
+            }
+
+            if (HasComp<DamageableComponent>(implanted))
+            {
```
Comment (EmoGarbage404) [inline Content.Server/Mindshield/MindShieldSystem.cs:70]: you don't need this check here

File: Content.Server/Mindshield/MindShieldSystem.cs [hunk 2]
```diff
@@ -21,6 +27,9 @@ public sealed class MindShieldSystem : EntitySystem
     [Dependency] private readonly MindSystem _mindSystem = default!;
     [Dependency] private readonly TagSystem _tag = default!;
     [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly BloodstreamSystem _bloodstreamSystem = default!;
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+    [Dependency] private readonly IPrototypeManager _Prototypes = default!;
```
Comment (EmoGarbage404) [inline Content.Server/Mindshield/MindShieldSystem.cs:32]: name should be _prototype

File: Content.Server/Mindshield/MindShieldSystem.cs [hunk 3]
```diff
@@ -52,6 +61,17 @@ public void MindShieldRemovalCheck(EntityUid implanted, EntityUid implant)
...
             _popupSystem.PopupEntity(Loc.GetString("head-rev-break-mindshield"), implanted);
             QueueDel(implant);
+            if (TryComp<BloodstreamComponent>(implanted, out var bloodstream))
+            {
+                _bloodstreamSystem.SpillAllSolutions(implanted, bloodstream);
+            }
+
+            if (HasComp<DamageableComponent>(implanted))
+            {
+                var damage = new DamageSpecifier(_Prototypes.Index<DamageGroupPrototype>("Genetic"), 300);
```
Comment (EmoGarbage404) [inline Content.Server/Mindshield/MindShieldSystem.cs:71]: don't hardcode the damage here. You should put this damage specifier either on a component somewhere or at the very least validate the prototype with a constant prototype Id validator in the system.

File: Resources/Locale/en-US/game-ticking/game-presets/preset-revolutionary.ftl [hunk 4]
```diff
@@ -14,7 +14,7 @@ head-rev-briefing =
     Use flashes to convert people to your cause.
     Kill all heads to take over the station.
 
-head-rev-break-mindshield = The Mindshield was destroyed!
+head-rev-break-mindshield = The Mindshield was destroyed! Their brain hemorrhages, Gross!.
```
Comment (EmoGarbage404) [inline Resources/Locale/en-US/game-ticking/game-presets/preset-revolutionary.ftl:17]: the wording on this is kinda undercutting the severity of it so i'd probably just reword it to "The Mindshield exploded violently!" or something like that.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]
I would still like to see changes that would make using mindshield implants less effective than killing conventionally.

**Comment by Titian3** [comment]
> my main concern is that this is an incredibly quick way to kill someone that is not communicated very well to the player. Instantly killing the player after a 3(?) second doafter is really extreme, especially for an antag. 

This was just my first thought to have a mechanical implementation to what most commonly happens during normal gameplay on rev rounds. So that additional rules are not required.
I'm keen for any other way this might solve the underlying problem, that being round stalling due to headrev protesting or sec RP'ing them as a prisoner to the determent of the rest of the players or a rule that just adds admin overhead to the prior.

Currently 1 of 2 things happen when a head rev is mindshielded, 
- Most commonly, the mind shield breaks, everyone sees and then whatever nearby player is around, all go to town and gib the head rev, If it is the last rev the round continues on to the end condition and shuttle is called.
- The Head rev is taken away by sec/captor, or the head rev demands a trial etc, essentially stalling and extending the round at a stage where a large portion of the players are ghosts already. Generally resulting in admin intervention to prevent stalling.

This would essentially do what happens already but prevent the stalling aspects of once a head rev is mindshielded.

>I could very much see these being used as a way to try and quick-kill revs. Stunning someone with a baton and quickly injecting while they're down could be seriously overpowered if it just straight up kills headrevs.

With regards to the power gaming example, people can do this already to make a determination of the head rev, they would just do the killing gibing in a subsequent action after seeing the mindshield broken text. If people were just randomly mindshielding it sounds more like a powergaming issue if people are running around beating random crew and doing a test mindshield.

File: Content.Server/Mindshield/MindShieldSystem.cs (hunk 3 above)
Comment (Titian3) [reply Content.Server/Mindshield/MindShieldSystem.cs]: I'm pretty sure I made the right changes here for what you mentioned, can you let me know if this is not correct. I moved it to a component and added that on to the MindShieldImplanter entity and reworked the system checks in a way that it could just be disabled back to current mechanic by removing that component off the MindShieldImplanter prototype.

**Comment by Titian3** [comment]
@EmoGarbage404 Thank you for the review, I appreciate it. I've made all the requested changes, although it seems like the vibe is not to go forward with this?

> my main concern is that this is an incredibly quick way to kill someone that is not communicated very well to the player. 

Just touching on this one too, the alternative feels much worse, there is not clear in game que that the head rev should be killed and then you have to go consult the wiki/rules/space law then go and find in some huge table for a particular action/problem that references another rule that says you should kill them without trial. 
https://wiki.spacestation14.io/wiki/Space_Law#Refusal_of_Mental_Shielding
Which feels like its even less communicated in game and a larger headache.

**Comment by Carolyn3114** [comment]
This also just promotes NRP and doesn't even allow for you to roleplay out execution/permaing of rev heads

**Comment by Titian3** [comment]
> This also just promotes NRP and doesn't even allow for you to roleplay out execution/permaing of rev heads

This is true a little, but at the benefit of the game being more self sufficient not requiring admin intervention, yet another rule added/managed, or the game mode breaking down because someone wants to make half the server of ghosts wait for a trial. 
The situation you speak of is exactly what I'm trying to prevent as when you get to late game revs it is inevitable you have at least several ghost and the round mode objective is that the revs or the heads die, not put them in jail or give them their day in court.

The space law/rule has already been updated to this for higher RP environments so the intent from the rules and game mode is that you should kill them without any long winded RP anyway. This just mechanically enforces this so some admin doesn't get saying this round is stalled because Urist mchands is stalling the round by putting the rev in perma instead of killing them and not following the execution rule/law.

This follows similar mechanical solutions in team antags, Nukies essentially have the same thing with the acidifier and the shuttle auto call on all dead for their mode and zombies have stall prevention too with over X% zombie pop shuttle is called.

**Comment by chromiumboy** [comment]
The concensus was that instant round removal of head revs isn't a good way of solving these sort of problems
